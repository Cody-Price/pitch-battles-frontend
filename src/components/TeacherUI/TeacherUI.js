import React, { Component } from "react";
import "./TeacherUI.css";

import { createClassFetch } from "../../utilities/fetchCalls";

class TeacherUI extends Component {
  constructor() {
    super();

    this.state = {
      activePage: "",
      nameOfClass: ""
    };
  }

  handleChange = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  handleSubmit = async event => {
    event.preventDefault();
    try {
      const data = await createClassFetch(
        this.state.nameOfClass,
        this.props.webToken
      );
      console.log(data);
    } catch (error) {
      console.log(error);
    }
  };

  render() {
    return (
      <main className="teacher-ui">
        hello teacher
        <form className="new-class-form" onSubmit={e => this.handleSubmit(e)}>
          <input
            type="text"
            name="nameOfClass"
            value={this.state.nameOfClass}
            onChange={e => this.handleChange(e)}
            placeholder="name of new class"
          />
          <input type="submit" />
        </form>
      </main>
    );
  }
}

export default TeacherUI;
