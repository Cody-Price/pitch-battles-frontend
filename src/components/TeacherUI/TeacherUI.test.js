import React from 'react';
import ReactDOM from 'react-dom';
import TeacherUI from './TeacherUI';
import Enzyme, { shallow } from 'enzyme';
const EnzymeAdapter = require('enzyme-adapter-react-16');

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe('TeacherUI', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<TeacherUI />)
	});

	it('should match the snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

})