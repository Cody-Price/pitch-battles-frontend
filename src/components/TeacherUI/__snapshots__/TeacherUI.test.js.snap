// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeacherUI should match the snapshot 1`] = `
<main
  className="teacher-ui"
>
  hello teacher
  <form
    className="new-class-form"
    onSubmit={[Function]}
  >
    <input
      name="nameOfClass"
      onChange={[Function]}
      placeholder="name of new class"
      type="text"
      value=""
    />
    <input
      type="submit"
    />
  </form>
</main>
`;
